flink-end-to-end-perf-tests/flink-basic-operations/src/main/java/org/apache/flink/basic/operations/PerformanceTestJob.java

```diff
@@ -0,0 +1,225 @@
+	private static String recordValue;
+
+	protected DataStream<Tuple2<String, String>> generateGraph(
+			String sourceName,
+			StreamExecutionEnvironment env,
+			int recordSize,
+			long maxCount,
+			int sleepNum,
+			int seed,
+			int parallism,
+			String streamPartitioner,
+			String outputPath){
+		if (recordSize == 10){
+			recordValue = recordSize10B;
+		} else if (recordSize == 100){
+			recordValue = recordSize100B;
+		} else {
+			recordValue = recordSize1KB;
+		}
+
+		DataStream<String> sourceNode = FromElements.fromElements(env, maxCount, sleepNum,
+			WordCountData.getWords(recordValue.length(), true, seed)).setParallelism(parallism).name(sourceName);
+
+		DataStream<Tuple2<String, String>> flapNode =
+			sourceNode
+				.flatMap((String value, Collector<Tuple2<String, String>> collector)
+					-> collector.collect(new Tuple2<String, String>(value, recordValue)))
+				.returns(Types.TUPLE(Types.STRING, Types.STRING))
+				.setParallelism(parallism)
+				.keyBy(0)
+				.flatMap(new ValueStateFlatMap());
+
+		DataStream<Tuple2<String, String>> outputNode = null;
+
+		if ("forward".equals(streamPartitioner)) {
+			outputNode = flapNode.forward();
+		} else if ("rescale".equals(streamPartitioner)) {
+			outputNode = flapNode.rescale();
+		} else if ("rebalance".equals(streamPartitioner)) {
+			outputNode = flapNode.rebalance();
+		} else if ("shuffle".equals(streamPartitioner)) {
+			outputNode = flapNode.shuffle();
+		} else if ("keyby".equals(streamPartitioner)) {
+			outputNode = flapNode.keyBy(0);
+		} else if ("broadcast".equals(streamPartitioner)) {
+			outputNode = flapNode.broadcast();
+		} else {
+			throw new IllegalArgumentException("Argument streamPartitioner is illegal!");
+		}
+		return outputNode;
+
+	}
+
+	protected DataStream<Tuple3<String, String, String>> connector(
+		DataStream<Tuple2<String, String>> output1,
+		DataStream<Tuple2<String, String>> output2,
+		int parallelism){
+		DataStream<Tuple3<String, String, String>> output = output1.connect(output2).process(
+			new CoProcessFunction<Tuple2<String, String>, Tuple2<String, String>, Tuple3<String, String, String>>() {
+				@Override
+				public void processElement1(Tuple2<String, String> value, Context ctx, Collector<Tuple3<String, String, String>> out) throws Exception {
+					out.collect(new Tuple3<>(value.f0, value.f1, value.f1));
+				}
+
+				@Override
+				public void processElement2(Tuple2<String, String> value, Context ctx, Collector<Tuple3<String, String, String>>out) throws Exception {
+					processElement1(value, ctx, out);
+				}
+
+			}, new TupleTypeInfo<>(BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO,
+				BasicTypeInfo.STRING_TYPE_INFO)).name("downStreamWithInputOrder").setParallelism(parallelism);
+		return output;
+	}
+
+	public static void main(String[] args) throws Exception {
+
+		ParameterTool paramTools = ParameterTool.fromArgs(args);
+		String jobName = paramTools.get("jobName", "Basic PerformanceTestJob");
```

Comment:
we _could_ use ConfigOptions for all of this. use `ParameterTool#toMap` to populate a `Configuration`, and then retrieve them via a set of static final `ConfigOptions`. Main benefit is that we can then auto-generate usage instructions for this job.

Reply:
ok, i update it